packages/conf/conf.go

```diff
@@ -0,0 +1,233 @@
+var Installed bool
+
+// Config - global immutable parameters
+var Config = *initialValues()
+
+func initialValues() *SavedConfig {
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.WithFields(log.Fields{"type": consts.IOError, "error": err}).Error("Getcwd failed")
+	}
+
+	return &SavedConfig{
+		LogLevel:     "INFO",
+		InstallType:  "PRIVATE_NET", // ??? PUBLIC_NET ???
+		NodeStateID:  "*",
+		StartDaemons: "",
+		//
+		TCPServer: HostPort{Host: "127.0.0.1", Port: 7078},
+		HTTP:      HostPort{Host: "127.0.0.1", Port: 7079},
+		StatsD:    StatsDConfig{Name: "apla", HostPort: HostPort{Host: "127.0.0.1", Port: 8125}},
+
+		DB: DBConfig{
+			Name:     "apla",
+			HostPort: HostPort{Host: "127.0.0.1", Port: 5432},
+			User:     "",
+			Password: "",
+		},
+
+		WorkDir:    cwd,
+		PrivateDir: "",
+
+		Centrifugo: CentrifugoConfig{
+			Secret: "",
+			URL:    "",
+		},
+	}
+
+}
+
+// GetConfigPath returns path from command line arg or default
+func GetConfigPath() string {
+	if *ConfigPath != "" {
+		return *ConfigPath
+	}
+	return filepath.Join(Config.WorkDir, defaultConfigFile)
+}
+
+// GetPidFile returns path to pid file
+func GetPidFile() string {
+	return filepath.Join(Config.WorkDir, "apla.pid")
+}
+
+// LoadConfig from configFile
+// the function has side effect updating global var Config
+func LoadConfig() error {
+	_, err := toml.DecodeFile(GetConfigPath(), &Config)
+	return err
+}
+
+// SaveConfig save global parameters to configFile
+func SaveConfig() error {
+	cf, err := os.Create(GetConfigPath())
+	if err != nil {
+		log.WithFields(log.Fields{"type": consts.IOError, "error": err}).Error("Create config file failed")
+		return err
+	}
+	defer cf.Close()
+	return toml.NewEncoder(cf).Encode(Config)
+}
+
+// NoConfig - config file does not exist
+func NoConfig() bool {
+	_, err := os.Stat(GetConfigPath())
+	return os.IsNotExist(err)
+}
+
+func flagOrEnv(confValue *string, flagValue string, envName string) {
+	if flagValue != "" {
+		*confValue = flagValue
+		return
+	}
+	if env, ok := os.LookupEnv(envName); ok {
+		*confValue = env
+	}
+}
+
+func intFlagOrEnv(confValue *int, flagValue int, envName string) {
+	if flagValue != 0 {
+		*confValue = flagValue
+		return
+	}
+	if env, ok := os.LookupEnv(envName); ok {
+		i, err := strconv.Atoi(env)
+		if err != nil {
+			log.WithFields(
+				log.Fields{"type": consts.ConfigError, "error": err},
+			).Error("Incorrect value in environment: " + envName)
+			return
+		}
+		*confValue = i
+	}
+}
+
+// OverrideFlags override default config values by environment or args
+func OverrideFlags() {
+
+	flagOrEnv(&Config.DB.Name, *FlagDbName, "PGDATABASE")
+	flagOrEnv(&Config.DB.Host, *FlagDbHost, "PGHOST")
+	intFlagOrEnv(&Config.DB.Port, *FlagDbPort, "PGPORT")
+	flagOrEnv(&Config.DB.User, *FlagDbUser, "PGUSER")
+	flagOrEnv(&Config.DB.Password, *FlagDbPassword, "PGPASSWORD")
+
+	// tcp
+	if *FlagTCPHost != "" {
+		Config.TCPServer.Host = *FlagTCPHost
+	}
+	if *FlagTCPPort != 0 {
```

Comment:
all config fields should be overrided in one way. Use falgOrEnv(&Config.TCPServer.Host, *FlagTCPHost, "") etc.